### DISCUSSION TITLE
RedwoodJs Tutorial vs. Redwood Chakra UI Generator results

### DISCUSSION THREAD

**Discussion by Philzen** [chakra-ui/chakra-ui#5785] [General]
When i create a Chakra-UI enabled redwood app via the `yarn rw setup ui chakra-ui` (which is an integration that was [recently added to redwood](https://github.com/redwoodjs/redwood/pull/3715) by a Chakra UI team member) i get the following structure in `App.tsx`:

```jsx
const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <RedwoodProvider titleTemplate="%PageTitle | %AppTitle">
      <ColorModeScript />
      <ChakraProvider>
        <RedwoodApolloProvider>
          <Routes />
        </RedwoodApolloProvider>
      </ChakraProvider>
    </RedwoodProvider>
  </FatalErrorBoundary>
)
```

The [Chakra UI RedwoodJS "Getting Started" tutorial](https://chakra-ui.com/guides/getting-started/redwoodjs-guide) however advises this structure:

```jsx
const App = () => (
  <ChakraProvider>
    <FatalErrorBoundary page={FatalErrorPage}>
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </FatalErrorBoundary>
  </ChakraProvider>
)
```

This makes me wonder:

1. What is the functional difference by using this different nesting (i.e. having the ErrorBoundary the most outer element)
2. Shouldn't the Chakra UI docs somehow be synchronized with the result of `yarn rw setup ui chakra-ui` and also refer to that as the best practice option?

**Comment by Philzen** [comment]
@TimKolberger Would you be able to shed some light on this?

**Reply by TimKolberger** [reply]
Having the FatalErrorBoundary as the outer most element, it could catch a potential error in the ChakraProvider.

I'd go for the first solution.

And the docs should be synchronized - kindly create a PR to the chakra docs if you have time 😊

**Reply by Philzen** [reply]
@TimKolberger thank you for the clarification. I created a PR for the docs in order to avoid future confusion for other users: https://github.com/chakra-ui/chakra-ui-docs/pull/492
